rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1197 @@
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.store.spi.DirectoryHelper.getVerifiedIndexDir(String, Properties, boolean) -->
+        <!-- use DirectoryHelper.getVerifiedIndexPath(java.lang.String, java.util.Properties, boolean) -->
+        <rule id="hsearch-00219">
+            <when>
+                <javaclass references="org.hibernate.search.store.spi.DirectoryHelper.getVerifiedIndexDir{*}">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated method DirectoryHelper#getVerifiedIndexDir in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>The `DirectoryHelper#getVerifiedIndexDir(String, Properties, boolean)` method is deprecated. Use `DirectoryHelper#getVerifiedIndexPath(java.lang.String, java.util.Properties, boolean)` instead. </message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_hibernate_search_advanced_integrator_changes"
+                        title="Advanced Integrator Changes in Hibernate Search 5.x" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.cfg.EntityDescriptor.setCacheInMemory(Map<String, Object>) -->
+        <!-- This will be removed with no replacement. -->
+        <rule id="hsearch-00220">
+            <when>
+                <javaclass references="org.hibernate.search.cfg.EntityDescriptor.setCacheInMemory{*}">
+                    <location>METHOD_CALL</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated method EntityDescriptor#setCacheInMemory in Hibernate Search 5" effort="1" severity="mandatory">
+                    <message>Method `org.hibernate.search.cfg.EntityDescriptor.setCacheInMemory` is deprecated and there is no replacement.</message>
+                    <link href="http://hibernate.org/search/documentation/migrate/5.5/#api-changes" title="Hibernate Search 5.4 to 5.5 migration" />
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscellaneous Hibernate Search Changes" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- org.hibernate.search.MassIndexer.threadsForSubsequentFetching(int) -->
+        <!-- Being ignored: this method will be removed. -->
+        <rule id="hsearch-00221">
+            <when>
+                <javaclass references="org.hibernate.search.MassIndexer.threadsForSubsequentFetching{*}">
```

Comment:
There's also a deprecated method `threadsForIndexWriter` in EAP 6's Hibernate Search. I don't see a rule for that, presumably because that method was already removed in EAP 7's Hibernate Search. WDYT about adding a rule? (I don't even want to guess how many similar cases we miss...)